gematria/datasets/basic_block_utils.h

```diff
@@ -28,19 +28,16 @@ namespace gematria {
 
 using namespace llvm;
 
-// Basic Block Utilities.
-class BasicBlockUtils {
- public:
-  // Gets the registers used by a basic block (in the form of a sequence of
-  // instructions).
-  static std::vector<unsigned> getUsedRegisters(
-      const std::vector<DisassembledInstruction> &Instructions,
-      const MCRegisterInfo &RegisterInfo, const MCInstrInfo &InstructionInfo);
-
-  static std::optional<unsigned> getLoopRegister(
-      const std::vector<DisassembledInstruction> &Instructions,
-      const MCRegisterInfo &RegisterInfo, const MCInstrInfo &InstructionInfo);
-};
+// Gets the registers used by a basic block (in the form of a sequence of
+// instructions).
+std::vector<unsigned> getUsedRegisters(
+    const ArrayRef<DisassembledInstruction> Instructions,
+    const MCRegisterInfo &RegisterInfo, const MCInstrInfo &InstructionInfo);
+
+std::optional<unsigned> getLoopRegister(
```

Comment:
Please add a comment explaining what this does (returns a GPR that is not touched by the basic block). The name might sound like we're looking for some kind of counter used by the basic block.

It could also make sense to rename this to `getUnusedGPRegister()` to make it more clear what this does (instead of naming it by what it is used for).